atomic_reactor/plugins/post_group_manifests.py

```diff
@@ -318,62 +154,27 @@ def sort_annotations(self):
 
-        sorted_digests = {}
-
-        for plat, annotation in all_annotations.items():
-            for digest in annotation['digests']:
-                hostname = registry_hostname(digest['registry'])
-                media_type = get_manifest_media_type(digest['version'])
-                if media_type not in self.manifest_media_types:
-                    continue
-
-                platforms = sorted_digests.setdefault(hostname, {})
-                repos = platforms.setdefault(plat, [])
-                repos.append(digest)
-
-        sources = {}
-        for registry in self.registries:
-            hostname = registry_hostname(registry)
-            platforms = sorted_digests.get(hostname, {})
-
-            if set(platforms) != all_platforms:
-                raise RuntimeError("Missing platforms for registry {}: found {}, expected {}"
-                                   .format(registry, sorted(platforms), sorted(all_platforms)))
-
-            selected_digests = {}
-            for p, repos in platforms.items():
-                selected_digests[p] = sorted(repos, key=lambda d: d['repository'])[0]
-
-            sources[registry] = selected_digests
-
-        return sources
-
-    def get_registry_session(self, registry):
-        registry_conf = self.registries[registry]
-
-        insecure = registry_conf.get('insecure', False)
-        secret_path = registry_conf.get('secret')
-
-        return RegistrySession(registry, insecure=insecure,
-                               dockercfg_path=secret_path,
-                               access=('pull', 'push'))
+        return self.manifest_util.sort_annotations(all_annotations)
 
     def run(self):
         digests = dict()
+        list_json = []
+
         for registry, source in self.sort_annotations().items():
-            session = self.get_registry_session(registry)
+            session = self.manifest_util.get_registry_session(registry)
 
             if self.group:
-                repo, digest = self.group_manifests_and_tag(session, source)
+                repo, digest, list_json = self.group_manifests_and_tag(session, source)
                 self.log.debug("repo: %s digest: %s", repo, digest)
                 digests[repo] = digest
             else:
                 found = False
                 if len(source) != 1:
                     raise RuntimeError('Without grouping only one source is expected')
                 for digest in source.values():
-                    self.tag_manifest_into_registry(session, digest)
+                    self.manifest_util.tag_manifest_into_registry(session, digest,
+                                                                  self.workflow.tag_conf.images)
                     found = True
                 if not found:
                     raise ValueError('Failed to find any platform')
-        return digests
+        return {"digests": digests, "list_json": list_json}
```

Comment:
The digest will be the same for every repository. Maybe we should really be returning "digest" and "repos" instead of "digests".

Reply:
Fixed.